python/lsst/obs/decam/decamMapper.py

```diff
@@ -263,16 +265,30 @@ def std_raw(self, item, dataId):
         result : `lsst.afw.image.Exposure`
             The standardized Exposure.
         """
-        # Convert the raw DecoratedImage to an Exposure, set metadata and wcs.
-        exp = exposureFromImage(item, logger=self.log)
-        md = exp.getMetadata()
-        visitInfo = self.makeRawVisitInfo(md=md)
-        exp.getInfo().setVisitInfo(visitInfo)
-
-        # Standardize an Exposure, including setting the photoCalib object
-        return self._standardizeExposure(self.exposures['raw'], exp, dataId,
+        return self._standardizeExposure(self.exposures['raw'], item, dataId,
                                          trimmed=False)
 
+    def _createInitialSkyWcs(self, exposure):
+        # DECam has a coordinate system flipped on X with respect to our
+        # VisitInfo definition of the field angle orientation.
+        self._createSkyWcsFromMetadata(exposure)
+
+        if exposure.getInfo().getVisitInfo() is None:
+            msg = "No VisitInfo; cannot access boresight information. Defaulting to metadata-based SkyWcs."
+            self.log.warn(msg)
+            return
+        try:
+            newSkyWcs = createInitialSkyWcs(exposure.getInfo().getVisitInfo(), exposure.getDetector(),
+                                            flipX=True)
+            exposure.setWcs(newSkyWcs)
+        except InitialSkyWcsError as e:
+            msg = "Cannot create SkyWcs using VisitInfo and Detector, using metadata-based SkyWcs: %s"
+            self.log.warn(msg, e)
```

Comment:
This is a lot of very specific error message handling. Can you give me a little context about why all this is useful here (and, perhaps more to the point, not used elsewhere as far as I know)?

Reply:
This method is identical to `obs.base._createInitialSkyWcs` (which it overrides), except for the `flipX=True` in `createInitialSkyWcs`. Unfortunately, I don't think there's a better approach until RFC-605 is implemented. I'll add a note to that effect, though.